File: python/6kyu/Character_with_longest_consecutive_repetition.py
def longest_repetition(chars):
    
    if len(chars) == 0:
        return '', 0
        
    the_char = chars[0]
    the_count = 1
    
    count = 1
    for i in range(1,len(chars)):
        
        if chars[i-1] == chars[i]:
            count += 1
        
            if the_count < count:
                the_char = chars[i]
                the_count = count
        else:
            count = 1
    
    return the_char, the_count

File: rust/src/6kyu/Character_with_longest_consecutive_repetition.rs
fn longest_repetition(s: &str) -> Option<(char, usize)> {
    if s.len() == 0 {
        return None;
    }

    let len = s.len();
    let mut count = 0;
    let mut curr_count = 1;
    let mut res: u8 = 0;

    s.chars().enumerate().for_each(|(i, c)| {
        if i < len - 1 && c == s.chars().nth(i + 1).unwrap() {
            curr_count += 1;
        } else {
            if curr_count > count {
                count = curr_count;
                res = c as u8;
            }
            curr_count = 1;
        }
    });

    Some((res as char, count))
}
